### PR TITLE
APC file cleaning, less hardcodes, more fixes

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/Monkestation/MonkeStation/pull/400
- https://github.com/tgstation/tgstation/pull/61655
- https://github.com/tgstation/tgstation/pull/69597
- https://github.com/tgstation/tgstation/pull/68835

## About The Pull Request
I took Monke's rendition of TG's APC deep cleaning, but also added later fixes they did not include.

This PR is intended to be a continuation of #8697 , as such, that PR should be merged prior to this one.

This PR is also intended to be merged in a similiar timeframe to #8694 , either can go first it doesnt really matter

I reviewed /tg/'s current live build and did the changes from that, so there shouldn't be any complications with the emissive system

> Cleans the code by using early returns and changing some var names to the appropriate ones and a spelling mistake.
> Add local APC cache to /area
> APC Defines moved to _DEFINES/power/apc_defines.dm
> Moved to _DEFINES/power.dm to _DEFINES/power/power.dm
> Moved GLOBAL_VAR_INIT(CHARGELEVEL, 0.001) and GLOBAL_LIST_EMPTY(powernets) from code/_globalvars/misc.dm to _DEFINES/power/power.dm.
> Added Mapping directional helpers for future use for wall objects like APCs
> Added some new sounds for switches
> Different power off sounds at energy 30% (equipment turn off) and 15% (light turn off) and 1% (all turn off).
> Power On sound for rebooting and charging back from 0%

> - When power fails, the apc update() proc gets called every timer tick, which is unnecessary. This moves the call to update() to energy_fail() because it technically only needs to be called once at the start.

Cleaned up some AI code and malf code. Pretty minor.

## Why It's Good For The Game

APC organization for easier reading/maintaining and better feedback with sounds when interacting with them.

The cleaning code also included a few more checks to reduce runtimes or bad calls.

Cleaned up malf code to have less hardcodes, cleaned up ai code to have less hardels.

image

I think this image speaks for itself

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

After defining, power switches still change according to power level
powerchanges accordingly

Malf menu appears correctly
menu

Correct change in behavior after hacking
shutncore

</details>

## Changelog
:cl:  RKz, DoctorSquishy, Ghilker, esainane, ShizCalev
code: /area now retains a local reference to the APC in the area, if any
code: combined all apc defines into one .dm
soundadd: New sounds for switches on APCs and power level on and off stages
refactor: Refactored and Organized APC code
code: less hardels in ai code
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
linter on holiday smh

**Comment by Tsar-Salat** [comment]
broke APCs detecting malfs.

fixing

**Comment by itsmeow** [comment]
Is this PR ready for review?

**Comment by Tsar-Salat** [comment]
> Is this PR ready for review?

broken malf detection atm

everything else looks fine. Sounds work.

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -46,6 +46,8 @@
 	/// This gets overridden to 1 for space in area/Initialize(mapload).
 	var/always_unpowered = FALSE
 
+	var/obj/machinery/power/apc/apc = null
```
Comment (PowerfulBacon) [inline code/game/area/areas.dm:49]: Where is this actually assigned to?

Comment (Tsar-Salat) [reply code/game/area/areas.dm:49]: Hm?

Its called about 19 times.

The purpose is to hold a reference to the local apc. If there is no local apc, the original value is never changed, and stays null.

Its a better alternative(imo) to the `get_apc()` proc, which was removed in the port.

File: code/modules/power/apc/apc_malf.dm [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 		return
 	if(malf.malfhacking)
-		to_chat(malf, "You are already hacking an APC.")
+		to_chat(malf, "<span class='warning'>You are already hacking an APC.</span>")
 		return
-	to_chat(malf, "Beginning override of APC systems. This takes some time, and you cannot perform other actions during the process.")
+	to_chat(malf, "<span class='notice'>Beginning override of APC systems. This takes some time, and you cannot perform other actions during the process.</span>")
 	malf.malfhack = src
-	malf.malfhacking = addtimer(CALLBACK(malf, TYPE_PROC_REF(/mob/living/silicon/ai, malfhacked), src), 600, TIMER_STOPPABLE)
+	malf.malfhacking = addtimer(CALLBACK(malf, TYPE_PROC_REF(/mob/living/silicon/ai/, malfhacked), src), 600, TIMER_STOPPABLE)
```
Comment (PowerfulBacon) [inline code/modules/power/apc/apc_malf.dm]:
```suggestion
	malf.malfhacking = addtimer(CALLBACK(malf, TYPE_PROC_REF(/mob/living/silicon/ai, malfhacked), src), 600, TIMER_STOPPABLE)
```

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
yolo

**Pull request merged**